src/commands/inline-activation.ts

```diff
@@ -0,0 +1,31 @@
+import * as vscode from "vscode";
+import { LanguageService } from "../kafka-file/languageservice/kafkaFileLanguageService";
+
+
+export class InlineCommandActivationCommandHandler {
+    public static commandId = 'vscode-kafka.inline.activate';
+
+    constructor(private readonly languageService: LanguageService) {}
+
+    async execute(): Promise<void> {
+        const editor = vscode.window.activeTextEditor;
+        const document = editor!.document;
+        const curPos = editor!.selection.active;
+        
+        if (document.languageId !== 'kafka') {
+            return;
+        }
+        const kafkaFileDocument = this.languageService.parseKafkaFileDocument(document);
```

Comment:
I think it is very shame 

 * to parse the full kafka file document to know which block is selected by the current position of the cursor.
 * to execute the full codelens to get the command.

I think your command should belong to the kafka file client https://github.com/jlandersen/vscode-kafka/blob/06b74ed03a57c73b2daec4a60a3dfd76145c039e/src/kafka-file/kafkaFileClient.ts#L115 (instead of registering your commandin the extension.ts, you register it in this kafka ckient file which is able to get the kafka file document).

 * step 1 :  get the kafka file document : I suggest that you do like codelens https://github.com/jlandersen/vscode-kafka/blob/06b74ed03a57c73b2daec4a60a3dfd76145c039e/src/kafka-file/kafkaFileClient.ts#L265 to get (or parse if need) the kafka file document.

 * step 2 : get the block which embedds the current position : to get the node from the given position, you can do like completion https://github.com/jlandersen/vscode-kafka/blob/06b74ed03a57c73b2daec4a60a3dfd76145c039e/src/kafka-file/languageservice/services/completion.ts#L24 (after that you can use parent to retrieve the block).

 * step3 : create the command for the block. I suggest that you move this function https://github.com/jlandersen/vscode-kafka/blob/06b74ed03a57c73b2daec4a60a3dfd76145c039e/src/kafka-file/languageservice/services/codeLensProvider.ts#L109 which create the producer command in the ProducerBlock. This method will be used by codelens and your `inlineCommandActivation.ts`

The code (should look like this:

```
class InlineCommandActivationCommandHandler extends AbstractKafkaFileFeature {

  public static commandId = 'vscode-kafka.inline.activate';

execute() {
        const editor = vscode.window.activeTextEditor;
        const document = editor!.document;
        const curPos = editor!.selection.active;

        if (document.languageId !== 'kafka') {
            return;
        }
        // parse or get the kafka file from the cache
        const kafkaFileDocument = this.getKafkaFileDocument(document);
        this.languageService.executeInlineCommand(document, kafkaFileDocument);
}

```

and your `inlineCommandActivation.ts` should look like this:


export class InlineCommandActivation {

...

    async executeInlineCommand(document: TextDocument, kafkaFileDocument: KafkaFileDocument){

       // Get the AST node before the position where complation was triggered
        const node = kafkaFileDocument.findNodeBefore(position);
        if (!node) {
            return;
        }
        const currentBlock = // use node or node.parent to retrieve the block
        if (currentBlock === undefined) {
            return;
        }
       const command = currentBlock.createCommand(range, clusterId);
       const commandArguments = command?.arguments;
       vscode.commands.executeCommand(command.command, ...commandArguments)
    };